numpy/core/src/npysort/heapsort.cpp

```diff
@@ -382,212 +382,349 @@ npy_aheapsort(void *vv, npy_intp *tosort, npy_intp n, void *varr)
 /***************************************
  * C > C++ dispatch
  ***************************************/
-
+template NPY_NO_EXPORT int
+heapsort_<npy::bool_tag, npy_bool>(npy_bool *, npy_intp);
 NPY_NO_EXPORT int
 heapsort_bool(void *start, npy_intp n, void *NPY_UNUSED(varr))
 {
     return heapsort_<npy::bool_tag>((npy_bool *)start, n);
 }
+
+template NPY_NO_EXPORT int
+heapsort_<npy::byte_tag, npy_byte>(npy_byte *, npy_intp);
```

Comment:
I'm curious why you needed to add these.

Reply:
This forces instantiation of the template so that the symbol gets generated (and can be used by quicksort). This should disappear once the npy_sort refactoring is finished.